core/trino-main/src/main/java/io/trino/eventlistener/EventListenerConfig.java

```diff
@@ -41,4 +43,17 @@ public EventListenerConfig setEventListenerFiles(List<String> eventListenerFiles
                 .collect(toImmutableList());
         return this;
     }
+
+    @Min(1)
+    public int getMaxConcurrentQueryCompletedEvents()
+    {
+        return maxConcurrentQueryCompletedEvents;
+    }
+
+    @Config("event-listener.max-concurrent-query-completed-events")
```

Comment:
I don't think "concurrent" is a good name for this. It doesn't convey the right concept or suggest a good mental model. An event is a point in time concept, so concurrent over what period? I would suggest using the term "pending" instead.

Reply:
Pending might suggest they are queued somehow, but they are not. 

"concurrent completed events" metric is only bumped when there are multiple (multithreaded) calls to event listener at the same time. Even if there are 1000 query completed events scheduled to be run by single threaded executor, "concurrent completed events" will be at most 1. The backpressure in this case comes not from executor queue, but from max number of concurrent queries.

Reply:
I see. In that case, I'm not sure I understand what problem this PR is trying to solve. If an event listener cannot handle a certain concurrent load, it should limit (or discard) events internally.

Reply:
The problem are badly written custom event listeners. Usually they are not written by core Trino experts or concurrency experts which leads to issues.

Reply:
That seems like a problem for whoever is deploying those poorly written event listeners. I'm not sure I agree with Trino having to work around plugins that are not designed to handle how Trino works.

Also, as I described elsewhere, the way this is implemented in the PR, a poorly written listener will affect the ability to collect events in listeners that don't have those issues.